File: ADN.py
def transcribe_to_rna(dna_strand):
    rna_strand = ""

    for nucleotide in dna_strand:
        if nucleotide == 'G':
            rna_strand += 'C'
        elif nucleotide == 'C':
            rna_strand += 'G'
        elif nucleotide == 'T':
            rna_strand += 'A'
        elif nucleotide == 'A':
            rna_strand += 'U'
        else:
            # Caracter no valido
            print("Error: Caracter no válido en la cadena de ADN.")
            exit(1)

    return rna_strand


def main():
    continuar = True

    while continuar:
        # Solicitud ADN
        dna_strand = input("Ingrese la cadena de ADN (o 'Q' para salir): ")

        if dna_strand.upper() == 'Q':
            continuar = False
        else:
            
            rna_strand = transcribe_to_rna(dna_strand)

            # Mostrar resultado
            print("Resultado de la cadena transcrita:", rna_strand)


if __name__ == "__main__":
    main()

File: ADN.java
import java.util.Scanner;

public class ADN {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean continuar = true;

        while (continuar) {
            // Solicitar de cadena ADN
            System.out.print("Ingrese la cadena de ADN (o 'Q' para salir): ");
            String dnaStrand = scanner.nextLine();

            if (dnaStrand.equalsIgnoreCase("Q")) {
                continuar = false;
            } else {
               
                String rnaStrand = transcribeToRna(dnaStrand);

                // Mostrar resultado
                System.out.println("Resultado de la cadena transcrita: " + rnaStrand);
            }
        }
    }

    private static String transcribeToRna(String dnaStrand) {
        StringBuilder rnaStrand = new StringBuilder();

        for (char nucleotide : dnaStrand.toCharArray()) {
            switch (nucleotide) {
                case 'G':
                    rnaStrand.append('C');
                    break;
                case 'C':
                    rnaStrand.append('G');
                    break;
                case 'T':
                    rnaStrand.append('A');
                    break;
                case 'A':
                    rnaStrand.append('U');
                    break;
                default:
                    // Evitar caracteres no validos
                    System.out.println("Error: Caracter no válido en la cadena de ADN.");
                    System.exit(1);
            }
        }

        return rnaStrand.toString();
    }
}
